service/resource/configmap/resource.go

```diff
@@ -188,7 +189,61 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
-	return nil, nil, nil, nil
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	currentConfigMaps, err := toConfigMaps(currentState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	desiredConfigMaps, err := toConfigMaps(desiredState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	var configMapsToCreate interface{}
+	{
+		configMapsToCreate, err = r.GetCreateState(ctx, obj, currentState, desiredState)
+		if err != nil {
+			return nil, nil, nil, microerror.Mask(err)
+		}
+	}
+
+	var configMapsToDelete []*apiv1.ConfigMap
+	{
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out which config maps have to be deleted")
+
+		for _, currentConfigMap := range currentConfigMaps {
+			if !containsConfigMap(desiredConfigMaps, currentConfigMap) {
+				configMapsToDelete = append(configMapsToDelete, currentConfigMap)
+			}
+		}
+
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d config maps that have to be deleted", len(configMapsToDelete)))
+	}
```

Comment:
Just FYI, some more thoughts about this. The deletion state here is basically the same as in `GetDeleteState`, with one exception. While the logic in `GetDeleteState` is an algorithm with destructive implications, the algorithm here has constructive implications. During usual deletion we want to remove all state. Here we want to make sure the current state in the system stays as it is desired. This means the condition has to be negated. Desired state is kept, while undesired state is removed.

Reply:
This may be a sign that we don't need `GetCreateState` and `GetDeleteState`.

Reply:
In my comment above I started the discussion about `GetCreateState` and if we do not need it. This is still TBD. IMO we need `GetDeleteState` and the deletion process within updates because it is a constructive deletion mechanism during updates and not the destructive deletion during real deletions. One is to maintain system state and only the system state that is desired. This means we can clean up resources that are not intended. The other is to remove all resources to purge the system state completely. One is "constructive". The other is "destructive".